src/OpenTelemetry/ProviderExtensions.cs

```diff
@@ -38,5 +38,15 @@ public static Resource GetResource(this BaseProvider baseProvider)
 
             return Resource.Empty;
         }
+
+        /// <summary>
+        /// Gets the <see cref="Resource"/> associated with the <see cref="BaseProvider"/>.
+        /// </summary>
+        /// <param name="baseProvider"><see cref="BaseProvider"/>.</param>
+        /// <returns><see cref="Resource"/>if found otherwise <see cref="Resource.Empty"/>.</returns>
+        public static Resource GetDefaultResource(this BaseProvider baseProvider)
+        {
+            return ResourceBuilder.CreateDefault().Build();
```

Comment:
We could cache this and reuse? If an exporters keeps calling this, we dont want to recreate the same resource over and over.

Reply:
Is that possible with a static, extension method? I agree that this is inefficient, but then we would have to instantiate a ProviderExtensions class for it to hold instance fields right?

Reply:
Just FYI Jager & Zipkin should only call this once. They build process/localendpoint on the first batch.

Reply:
Yea but for any others.
I was wondering if it makes sense to make `Resource.Default` just like the Empty resource. (This will mean Resource.CreateDefault.Build() and Resource.Default would become 2 ways of achieving the same purpose)

Reply:
Given the immediate use case for this is from zipkin/jager, which does the caching, we can leave this as is. And do the perf optimization post 1.0

Reply:
@cijothomas Having...

```csharp
public class Resource
{
   public static Resource Default { get; } = new Resource(new Dictionary<string, object>
        {
            [ResourceSemanticConventions.AttributeServiceName] = "unknown_service"
                + (string.IsNullOrWhiteSpace(System.Diagnostics.Process.GetCurrentProcess().ProcessName)
                ? string.Empty :
                ":" + System.Diagnostics.Process.GetCurrentProcess().ProcessName),
        })
}
```

...sounds reasonable to me. I think that's what you are getting at 😄

Reply:
Yea this looks good. But there is ResourceBuilder().CreateDefault().Build() as well.. 2 ways to get to the same thing !